mungegithub/mungers/check-labels_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/google/go-github/github"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+var (
+	_                              = fmt.Printf
+	_                              = glog.Errorf
+	repoLabelNames []string        = []string{"team/abc", "team/def", "team/ghi"}
+	repoColors     []string        = []string{"d4c5f9", "fef2c0", "c7def8"}
+	fileLabelNames []string        = []string{"release-note", "dependency/rkt", "team/ghi"}
+	fileColors     []string        = []string{"d4c5f9", "fbfa04", "c7def8"}
+	repoLabels     []*github.Label = make([]*github.Label, 0, 0)
+	fileLabels     []*github.Label = make([]*github.Label, 0, 0)
+)
+
+type LabelAccessorTest struct {
+	AddedLabels []*github.Label
+	RepoLabels  []*github.Label
+}
+
+func (l *LabelAccessorTest) AddLabel(label *github.Label) error {
+	l.AddedLabels = append(l.AddedLabels, label)
+	return nil
+}
+
+func (l *LabelAccessorTest) GetLabels() ([]*github.Label, error) {
+	return l.RepoLabels, nil
+}
+
+func (l *LabelAccessorTest) GetFileContents(string) (string, string, error) {
+	return `- color: d4c5f9
+  name: release-note
+- color: fbfa04
+  name: dependency/rkt
+- color: c7def8
+  name: team/ghi`, "sha", nil
+}
+
+func makeCheckLabelsMunger() *CheckLabelsMunger {
+	return &CheckLabelsMunger{
+		labelAccessor: &LabelAccessorTest{},
+	}
+}
+
+func TestEachLoop(t *testing.T) {
+	runtime.GOMAXPROCS(runtime.NumCPU())
+
+	// setup some labels for the test.
+	for i := 0; i < len(fileLabelNames); i++ {
+		g := github.Label{Name: &fileLabelNames[i], Color: &fileColors[i]}
+		fileLabels = append(fileLabels, &g)
+	}
+
+	for i := 0; i < len(repoLabelNames); i++ {
+		g := github.Label{Name: &repoLabelNames[i], Color: &repoColors[i]}
+		repoLabels = append(repoLabels, &g)
+	}
+
+	tests := []struct {
+		name       string
+		repoLabels []*github.Label
+		expected   []*github.Label
+	}{
+		{
+			name:       "No labels in repository.",
+			repoLabels: []*github.Label{},
+			expected:   fileLabels,
+		},
+		{
+			name:       "Identical labels in repository and file.",
+			repoLabels: fileLabels,
+			expected:   []*github.Label{},
+		},
+		{
+			name:       "Adding new labels with existing labels in repository.",
+			repoLabels: repoLabels,
+			expected:   fileLabels[0:2],
+		},
```

Comment:
As far as I checked, the label name needs to be unique, or github won't let us add it, even with a different color. In check-labels.go:103, the name of the label is checked for uniqueness and we don't add it in the above case.